src/Timecode.php

```diff
@@ -63,11 +67,37 @@ public function __construct(int $hours, int $minutes, int $seconds, int $frames,
      */
     public static function createFromString(string $timecodeStr, float $framerate = self::DEFAULT_FRAMERATE) : self
     {
-        list($hours, $minutes, $seconds, $frames) = explode(':', $timecodeStr);
-
-        // @codingStandardsIgnoreStart
-        return new self((int) $hours, (int) $minutes, (int) $seconds, (int) $frames, $framerate);
-        // @codingStandardsIgnoreEnd
+        // Support "hours:minutes:seconds:frames" notation
+        $matches = [];
+        preg_match('@^(?P<hours>^\d{1,3}):(?P<minutes>\d{1,2}):(?P<seconds>\d{1,2}):(?P<frames>\d{1,2})$@', $timecodeStr, $matches);
+        if (!empty($matches['hours']) && !empty($matches['minutes']) && !empty($matches['seconds']) && !empty($matches['frames'])) {
+            // @codingStandardsIgnoreStart
+            return new self(
+                $matches['hours'],
+                $matches['minutes'],
+                $matches['seconds'],
+                $matches['frames'],
+                $framerate
+            );
+            // @codingStandardsIgnoreEnd
+        }
+
+        // Support "hours:minutes:seconds.milliseconds" notation
+        $matches = [];
+        preg_match('@^(?P<hours>^\d{1,3}):(?P<minutes>\d{1,2}):(?P<seconds>\d{1,2})\.(?P<ms>\d{1,3})$@', $timecodeStr, $matches);
```

Comment:
sorry, I'm not sure it's a good idea to do so for 2 reasons:
- I think it will provide a better developer experience to have only 1 method to use for any kind of Timecode notation (something like `new DateTime('now')`, `new DateTime('2017-02-02')`, `new DateTime('2017-02-02' 12:43:24)`, etc.)
- If we make another method to handle this new string notation, we'll end up with a method named like `createFromStringMillisecondNotation` then we'll also probably want to change `createFromString` to `createFromStringFrameNotation` to be consistent, which will be a big breaking change :(

I propose to wait for another opinion